### PR TITLE
python3Packages.daff: init at 1.3.46

### PR DESCRIPTION
**Pull request by turion**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by turion** [comment]
One thing I'm surprised about:

```
❯ nix shell .#python3Packages.daff
❯ daff
Traceback (most recent call last):
  File "/nix/store/h4238v7vy9q9hv7j0wk8fzixk1nkak97-python3.11-daff-1.3.46/bin/.daff-wrapped", line 6, in <module>
    from daff import main
  File "/nix/store/h4238v7vy9q9hv7j0wk8fzixk1nkak97-python3.11-daff-1.3.46/bin/daff.py", line 2
    PATH=${PATH:+':'$PATH':'}
         ^
SyntaxError: invalid syntax
❯ daff.py 
daff can produce and apply tabular diffs.
Call as:
  daff a.csv b.csv
  daff [--color] [--no-color] [--output OUTPUT.csv] a.csv b.csv
  daff [--output OUTPUT.html] a.csv b.csv
  daff [--www] a.csv b.csv
  daff parent.csv a.csv b.csv
  daff --input-format sqlite a.db b.db
  daff patch [--inplace] a.csv patch.csv
[...]
```
What do I need to do such that the `daff` executable is interpreted correctly (i.e. not by python)?

**Comment by turion** [comment]
CC @paulfitz

File: pkgs/development/python-modules/daff/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  propagatedBuildInputs = [
+  ];
+
+  pythonImportsCheck = [
+    "daff"
+  ];
+
+  meta = with lib; {
+    description = "data diff";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daff/default.nix:23]:
```suggestion
    description = "Library for comparing tables, producing a summary of their differences, and using such a summary as a patch file";
```

File: pkgs/development/python-modules/daff/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "1.3.46";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-ItDan9ajJ1tUySapyXsYD5JYqtZRE+oY8/7FLLrc2Bg=";
+  };
+
+  propagatedBuildInputs = [
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daff/default.nix:17]: We want to execute the tests if possible https://github.com/paulfitz/daff/tree/master/test

File: pkgs/development/python-modules/daff/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "daff";
+  version = "1.3.46";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-ItDan9ajJ1tUySapyXsYD5JYqtZRE+oY8/7FLLrc2Bg=";
+  };
+
+  propagatedBuildInputs = [
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daff/default.nix:16]:
```suggestion
```

File: pkgs/development/python-modules/daff/default.nix (hunk 2 above)
Comment (turion) [reply pkgs/development/python-modules/daff/default.nix:17]: I believe these tests are not related to the python package `daff`. They are for the companion JS package.

File: pkgs/development/python-modules/daff/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "daff";
+  version = "1.3.46";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-ItDan9ajJ1tUySapyXsYD5JYqtZRE+oY8/7FLLrc2Bg=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daff/default.nix:13]:
```suggestion
  };

  # there are no tests
  doCheck = false;
```

**Comment by SuperSandro2000** [comment]
@ofborg build python311Packages.daff

**Pull request merged**